### PR TITLE
sanity check on non conforming serial number of 0

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**
ZD 17757

### CODE REVIEW

File: wolfcrypt/src/asn.c [hunk 1]
```diff
@@ -21875,6 +21875,19 @@ static int DecodeCertInternal(DecodedCert* cert, int verify, int* criticalExt,
         /* Set fields extracted from data. */
         cert->version = version;
         cert->serialSz = (int)serialSz;
+
+    #if !defined(WOLFSSL_NO_ASN_STRICT) && !defined(WOLFSSL_PYTHON)
```
Comment (dgarske) [inline wolfcrypt/src/asn.c:21879]: I understand the no asn struct, but why the `WOLFSSL_PYTHON`? Does it specifically use a serial no 0 for testing?

File: tests/api.c [hunk 2]
```diff
@@ -53763,6 +53763,74 @@ static int test_MakeCertWithPathLen(void)
...
+
+#ifdef WOLFSSL_CERT_EXT
+    cert.keyUsage |= KEYUSE_KEY_CERT_SIGN;
+#endif
+
+    /* set serial number to 0 */
+    cert.serialSz  = 1;
+    cert.serial[0] = 0;
+
+    ExpectIntGE(wc_MakeCert(&cert, der, FOURK_BUF, NULL, &key, &rng), 0);
```
Comment (dgarske) [inline tests/api.c:53812]: Do we also want to cause an error on `wc_MakeCert` for this invalid serial?

File: wolfcrypt/src/asn.c (hunk 1 above)
Comment (JacobBarthelmeh) [reply wolfcrypt/src/asn.c:21879]: Exactly, it appears that one of the test certificates has a serial number of 0.

```
Python-3.8.14/Lib/test$ wolfssl x509 -serial -noout -in nullbytecert.pem
serial=00
```

Removing parsing of serial number 0 results in the equivalent "make check" failing with Python.

File: tests/api.c (hunk 2 above)
Comment (JacobBarthelmeh) [reply tests/api.c:53812]: Was leaning towards leaving it allowed to create non-conforming certificates. Could add in the check though.

**Review by dgarske — APPROVED** [review]
I'm okay with allowing it for now also.

**Pull request merged**